burgle.lic

```diff
@@ -13,77 +13,166 @@ class Burgle
   include DRCT
   include DRCI
 
-
   def initialize
 
     arg_definitions = [
       [
-        { name: 'start', regex: /start/i, description: 'Required: prevents accidentally running burgle and protects against burgle typo for starting script' },
+        { name: 'start', regex: /start/,optional: false, description: 'Required: prevents accidentally running burgle and protects against burgle typo for starting script' },
+        { name: 'entry', options: %w[lockpick rope cycle], optional: true, description: 'Override yaml setting for entry_type.' },
+        { name: 'roomid', regex: /\d+/, optional: true, description: 'Override yaml setting and go to room id (#) specified.  The room shold be chosen VERY carefully.'},
+        { name: 'loot_type', options: %w[drop keep], optional: true, description: 'Override yaml setting for loot. (items on item_whitelist are always kept.)' },
       ]
     ]
 
     args = parse_args(arg_definitions)
 
+    @entry_type = args.entry if args.entry
+    @burgle_room = args.roomid if args.roomid
+    @loot_type = args.loot_type if args.loot_type
+
+    #hash which maps directions to their reverse, for backing out of the house
+    @reverse_direction_map = {
+      'east' => 'west',
+      'west' => 'east',
+      'south' => 'north',
+      'north' => 'south',
+      'northeast' => 'southwest',
+      'southwest' => 'northeast',
+      'northwest' => 'southeast',
+      'southeast' => 'northwest'
+    }
+
+    #maps the searchable objects based on the room title
+    @room_searchable_objects_map = {
+      "Kitchen" => 'counter',
+      "Bedroom" => 'bed',
+      "Armory" => 'rack',
+      "Library" => 'bookshelf',
+      "Sanctum" => 'desk',
+      "Work Room" => 'table'
+    }
+
     start_script('jail-buddy') unless Script.running?('jail-buddy')
+
     @settings = get_settings
     @burgle_settings = @settings.burgle_settings
+    @loot_container = @burgle_settings['loot_container']
+    @use_lockpick_ring =  @burgle_settings['use_lockpick_ring']
+    @lockpick_container =  @burgle_settings['lockpick_container']
+
+    #set yaml settings unless being overridden from the command line arguments
+    @entry_type = @burgle_settings['entry_type'] unless @entry_type
```

Comment:
💡 I _think_ you can simplify this line to `@entry_type |= @burgle_settings['entry_type']`.

I _think_ you could instead simplify line 29 to `@entry_type = args.entry || @burgle_settings['entry_type']` (assuming `@burgle_settings` has already been defined).

Reply:
The first one doesn't seem to work when I tested.
I get an validation error (@entry_type gets set to a boolean).

```ruby
    @entry_type |= @burgle_settings['entry_type']
    echo @entry_type
```

```
,burgle start rope 
--- Lich: burgle active.
[burgle: true]
| Invalid burgle_settings:entry_type setting.
| It is very important that you check the documentation before running: https://elanthipedia.play.net/Lich_script_repository#burgle 
| This is a dangerous script to run if it's not understood.  If you get caught there are very high fines, and the loss of all your items is possible if you can't pay your debt.
```

The second does seem to work:
```ruby
    @entry_type = args.entry || @burgle_settings['entry_type']
    @burgle_room = args.roomid || @burgle_settings['room']
    @loot_type = args.loot_type ||  @burgle_settings['loot']
```